### PR TITLE
feat(replies): Reply via pop3 polling

### PR DESCRIPTION
**Pull request by merkata**
Note: this PR works in theory and does not currently ship with an integration test - reason being to test this, you need two mail servers (receiving mail into Discourse, configured server with pop3 enabled, sending mails to Discourse) and a mechanism to respond to a thread via e-mail and ensure it is received. Overall a painful and tricky business. Proposal is to test this manually.

Second note: if already configured via GUI, the changes can/will get overwritten when moving to env variables, that is by design, see [here](https://meta.discourse.org/t/s3-backup-bucket-option-missing-from-settings-backups/224802).

### Overview

This PR adds the capability to enable replies to threads via pop3 polling as described [here](https://meta.discourse.org/t/set-up-reply-by-email-with-pop3-polling/14003) via `juju config` on the parameters.

### Rationale

This capability is available via GUI and it was requested that it is available as a config as well.

### Juju Events Changes

none

### Module Changes

Adds new pop3 parameters to `config.yaml` and sets these as env variables to feed to pebble in `src/charm.py`.

### Library Changes

none

### Checklist

- [x ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x ] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x ] The documentation is generated using `src-docs`
- [x ] The documentation for charmhub is updated.
- [x ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by jdkandersson — APPROVED** [review]

File: config.yaml [hunk 1]
```diff
@@ -134,3 +134,27 @@ options:
...
+    default: ""
+  pop3_polling_password:
+    type: string
+    description: The password of that email account.
+    default: ""
+  pop3_polling_host:
+    type: string
+    description: Polling host such as pop.gmail.com.
+    default: ""
+  pop3_polling_enabled:
```
Comment (cbartz) [inline config.yaml:153]: What happens if this flag is enabled but the other configuration options are not provided? Shouldn't there be some sort of non-emptiness check? What does the GUI do in this case?

Comment (nrobinaubertin) [reply config.yaml:153]: Passing some environment variables has the same effect as using the GUI, so discourse will not enable the polling in that case but will not complain (we may see some alert in the admin settings)

Comment (nrobinaubertin) [reply config.yaml:153]: I agree that we should probably add some logic to put a warning in the status of the charm, or an error

**Review by amandahla — APPROVED** [review]

File: config.yaml (hunk 1 above)
Comment (merkata) [reply config.yaml:153]: We might think about this, though like Niels said - Discourse should handle that. The best we could do is to put a warning maybe, as you could configure these settings in any order and once you have them, polling will work. If not, we are guiding the user to set them all at once, like `juju config a=x b=y..` on one line - which is also an option I guess, then we need to update the documentation to reflect that this is how we expect the charm to be configured in terms of polling.

**Comment by merkata** [comment]
Original RT is closed, so I’m closing this one as well and it can be reopened if needed.

**Pull request closed without merging**